src/agents/customer_group.jl

```diff
@@ -1098,16 +1131,16 @@ function solve_agent_problem!(
             der_aggregator.incentive_level(reg_year_index_dera, z) * customers.Opti_DG(z, h, :BTMStorage) * der_aggregator.aggregation_level(reg_year_index_dera, z) -
             # cost of distributed generation 
             sum(customers.FOM_DG(z, h, m) * customers.Opti_DG(z, h, m) for m in customers.index_m)
-    end
-
+    end    
 
     # Calculate payback period of DPV-only
     # The NetProfit represents the energy saving/credit per representative agent per DER technology, 
     # assuming the optimal DER technology size
+    Payment_after_PVOnly = make_keyed_array(model_data.index_z, model_data.index_h)
     NetProfit_PV_only = make_keyed_array(model_data.index_z, model_data.index_h)
     for z in model_data.index_z, h in model_data.index_h
-        # value of distributed generation (offset load)
-        NetProfit_PV_only(z, h, :) .=
+        Payment_after_PVOnly(z, h, :) .= 
+            # value of distributed generation (offset load)
```

Comment:
This calculation for Payment_after_PVOnly is incorrect. Actual value is Payment_before_PVStor MINUS the value calculated here.